geomesa-convert/geomesa-convert-xml/src/main/scala/org/locationtech/geomesa/convert/xml/XMLConverter.scala

```diff
@@ -0,0 +1,83 @@
+package org.locationtech.geomesa.convert.xml
+
+import java.io.ByteArrayInputStream
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.xpath.{XPathConstants, XPathExpression, XPathFactory}
+
+import com.google.common.base.Charsets
+import com.typesafe.config.Config
+import org.locationtech.geomesa.convert.Transformers.{EvaluationContext, Expr}
+import org.locationtech.geomesa.convert._
+import org.opengis.feature.simple.SimpleFeatureType
+import org.w3c.dom.NodeList
+
+import scala.collection.JavaConversions._
+
+class XMLConverter(val targetSFT: SimpleFeatureType,
+                   val idBuilder: Expr,
+                   val featurePath: Option[XPathExpression],
+                   val inputFields: IndexedSeq[Field]) extends ToSimpleFeatureConverter[String] {
+
+  private val docBuilder = {
+    val factory = DocumentBuilderFactory.newInstance()
+    factory.setNamespaceAware(false)
```

Comment:
I added it as a separate step to validate the document if an xsd is present. I tried adding it to the parser itself, but then the xpaths have to have namespaces registered with them, and it becomes kind of a nightmare... hopefully this will be good enough.